src/vm/builtins/Array.zig

```diff
@@ -60,9 +61,34 @@ fn array_reverse(vm: *VirtualMachine, arguments: []const Code.Value) Code.Value
     return Code.Value.Object.Array.init(vm.allocator, new_array) catch Code.Value{ .none = {} };
 }
 
-fn length(vm: *VirtualMachine, arguments: []const Code.Value) Code.Value {
-    _ = vm;
+fn foreach(vm: *VirtualMachine, arguments: []const Code.Value) Code.Value {
+    if (!((arguments[0] == .object and (arguments[0].object == .array or arguments[0].object == .map)) and (arguments[1] == .object and arguments[1].object == .function))) {
+        return Code.Value{ .none = {} };
+    }
+
+    const callback = arguments[1].object.function;
+
+    if (arguments[0].object == .map) {
+        const map = arguments[0].object.map;
+        var map_iter = map.inner.iterator();
+
```

Comment:
map_entry_iterator

Reply:
its already self describable, `map_iter` would _probably_ translate to `map_iterator`, we later specify the type of iterator's items (`map_entery`)

Reply:
It is just more verbose and less devious, you could just say map.keyIterator or map.valueIterator and it is still map_iter, by asking the reader to read more for context is like sipping into a muddy water and at the end there is milk